src/main/groovy/edu/oregonstate/mist/metaxe/XEAppsResource.groovy

```diff
@@ -0,0 +1,139 @@
+package edu.oregonstate.mist.metaxe
+
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.api.jsonapi.ResourceObject
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+
+import com.codahale.metrics.annotation.Timed
+import javax.annotation.security.PermitAll
+import javax.ws.rs.GET
+import javax.ws.rs.NotFoundException
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.UriBuilder
+import java.util.regex.Pattern
+
+@Path("/xeapps")
+@Produces(MediaType.APPLICATION_JSON)
+@groovy.transform.TypeChecked
+class XEAppsResource extends Resource {
+    private XEAppDAO dao
+    private URI myEndpointUri
+
+    private final String JSONAPI_TYPE = "xeapp"
+
+    XEAppsResource(XEAppDAO dao, URI endpointUri) {
+        this.dao = dao
+        this.myEndpointUri = endpointUri
+        this.endpointUri = endpointUri
+    }
+
+    // Get a specific application by ID.
+    @Timed
+    @GET
+    @Path("{id}")
+    @PermitAll
```

Comment:
From looking at the docs this annotation is meant to specify what roles can access a given endpoint. Why not define this annotation at the class level once?